Towny/src/main/java/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -4177,6 +4215,25 @@ else if (town == trustTown) {
 		town.save();
 	}
 
+	private static void parseTownTrustTownRemoveAllCommand(CommandSender sender, Town town) throws TownyException {
+		List<Town> trustedTowns = new ArrayList<>(town.getTrustedTowns());
+
+		if (trustedTowns.isEmpty()) {
+			TownyMessaging.sendMsg(sender, Translatable.of("msg_no_trusted_towns"));
+			return;
+		}
+
+		for (Town trustTown : trustedTowns) {
+			BukkitTools.ifCancelledThenThrow(new TownTrustTownRemoveEvent(sender, trustTown, town));
+			town.removeTrustedTown(trustTown);
+			trustTown.getResidents().forEach(res -> plugin.deleteCache(res));
+		}
+

```

Comment:
I have pushed a commit that makes the parseTownTrustTownCommand have sub methods, you will have to rebase this PR to grab that commit. 

This will allow you to reuse the new remove method, similar to the request I made for the other command.

Also please catch the cancelled event exception and add a Confirmation.